include/cadmium/basic_model/partial_accumulator.hpp

```diff
@@ -0,0 +1,149 @@
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#ifndef CADMIUM_PARTIAL_ACCUMULATOR_HPP
+#define CADMIUM_PARTIAL_ACCUMULATOR_HPP
+#include<cadmium/modeling/ports.hpp>
+#include<cadmium/modeling/message_bag.hpp>
+#include<limits>
+
+
+namespace cadmium {
+    namespace basic_models {
+
+    /**
+     * @brief PartialAccumulator PDEVS Model.
+     *
+     * PartialAccumulator PDEVS Model:
+     * - X = Numeric
+     * - Y = Numeric
+     * - In_Ports: add<Numeric>, reset, partial
+     * - Out_Ports: outport
+     * - S = {Numeric:total, bool:reseted, int:partial}
+     * - internal({total, true, i}) = {0, false, 0}
+     *   internal({total, false, 1}) = {total, false, 0}
+     * - external({total, b, i}, t, x.add) = {total+x, b, i}
+     *   external({total, b, i}, t, x.reset) = {total, true, i}
+     *   external({total, b, i}, t, x.partial) = {total, b, 1}
+     * - confluence({total, b, i}, 0, x) = external(internal({total, b, i}), 0, x)
+     * - output ({total, b, 1}) = outport:{total}
+     * - time_advance({total, b, 1}) = 0
+     *   time_advance({total, true, i}) = 0
+     *   time_advance ({total, false, 0}) = infinite
+    */
+
+    template<typename VALUE, typename SET> //value is the type of accumulated values
+      struct partial_accumulator_defs {
+        struct sum : public out_port<VALUE> {
+        };
+        struct add : public in_port<VALUE> {
+        };
+        struct reset : public in_port<SET> {
+        };
+        struct partial : public in_port<SET>{
+        };
+    };   
+
+    template<typename VALUE, typename SET, typename TIME> //value is the type of accumulated values
+      struct partial_accumulator {
+        using on_reset=bool;
+        using on_partial=int;
+
+        using defs = partial_accumulator_defs<VALUE, SET>;
+
+        using state_type=std::tuple<VALUE, on_reset, on_partial>;
+        state_type state = std::make_tuple(VALUE{}, false, 0);
+
+        //default constructor
+        constexpr partial_accumulator() noexcept {}
+
+        //ports_definition
+        using input_ports=std::tuple<typename defs::add, typename defs::reset, typename defs::partial>;
+        using output_ports=std::tuple<typename defs::sum>;
+
+        // PDEVS functions
+        void internal_transition() {
+          if(!std::get<on_reset>(state) && std::get<on_partial>(state)!=1) {
+            throw std::logic_error("Internal transition called while not on reset or not on partial state");
+          }
+          if(std::get<on_reset>(state)) {

```

Comment:
If you make the change to use bool, in place of using get, you should just call the variable inside the struct:
if (std::get\<on_reset\>(state))
should be
if (state.on_reset)